src/lib.rs

```diff
@@ -1,744 +1,345 @@
-            IRShaderStage,
-            *mut IRMetalLibBinaryOpaque,
-        ) -> bool,
-    >,
-    get_reflection: libloading::Symbol<
-        'lib,
-        unsafe extern "C" fn(
-            *mut IRObjectOpaque,
-            IRShaderStage,
-            *mut IRShaderReflectionOpaque,
-        ) -> bool,
-    >,
-    get_type: libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRObjectOpaque) -> IRObjectType>,
-    serialize: libloading::Symbol<'lib, unsafe extern "C" fn() -> ()>,
-}
-
-pub struct IRObject<'lib> {
-    me: *mut IRObjectOpaque,
-    funcs: IRObjectFn<'lib>,
+pub struct IRObject {
+    me: *mut bindings::IRObject,
+    funcs: Arc<bindings::metal_irconverter>,
 }
 
-impl<'lib> Drop for IRObject<'lib> {
+impl Drop for IRObject {
     fn drop(&mut self) {
-        unsafe { (self.funcs.destroy)(self.me) }
+        unsafe { self.funcs.IRObjectDestroy(self.me) }
     }
 }
 
-impl<'lib> IRObject<'lib> {
+impl IRObject {
     pub fn create_from_dxil(
-        lib: &'lib libloading::Library,
+        compiler: &IRCompiler,
         bytecode: &[u8],
-    ) -> Result<IRObject<'lib>, Box<dyn std::error::Error>> {
+    ) -> Result<IRObject, Box<dyn std::error::Error>> {
         unsafe {
-            let funcs = IRObjectFn {
-                create_from_dxil: lib.get(b"IRObjectCreateFromDXIL")?,
-                destroy: lib.get(b"IRObjectDestroy")?,
-                get_metal_ir_shader_stage: lib.get(b"IRObjectGetMetalIRShaderStage")?,
-                get_metal_lib_binary: lib.get(b"IRObjectGetMetalLibBinary")?,
-                get_reflection: lib.get(b"IRObjectGetReflection")?,
-                get_type: lib.get(b"IRObjectGetType")?,
-                serialize: lib.get(b"IRObjectSerialize")?,
-            };
-
-            let me = (funcs.create_from_dxil)(
+            let me = compiler.funcs.IRObjectCreateFromDXIL(
                 bytecode.as_ptr(),
                 bytecode.len(),
-                IRBytecodeOwnership::None,
+                bindings::IRBytecodeOwnership::IRBytecodeOwnershipNone,
             );
 
-            Ok(Self { funcs, me })
+            Ok(Self {
+                funcs: compiler.funcs.clone(),
+                me,
+            })
+        }
+    }
+
+    pub fn gather_raytracing_intrinsics(&self, entry_point: &CStr) -> u64 {
+        unsafe {
+            self.funcs
+                .IRObjectGatherRaytracingIntrinsics(self.me, entry_point.as_ptr().cast())
         }
     }
 
     pub fn get_type(&self) -> IRObjectType {
-        unsafe { (self.funcs.get_type)(self.me) }
+        unsafe { self.funcs.IRObjectGetType(self.me.cast_const()) }
     }
 
     pub fn get_metal_ir_shader_stage(&self) -> IRShaderStage {
-        unsafe { (self.funcs.get_metal_ir_shader_stage)(self.me) }
+        unsafe { self.funcs.IRObjectGetMetalIRShaderStage(self.me) }
     }
 
     pub fn get_metal_lib_binary(
         &self,
         shader_stage: IRShaderStage,
         dest_lib: &mut IRMetalLibBinary,
     ) -> bool {
```

Comment:
Is this intended to be an error? And is this supposed to be a "new" library?

If so this should return `Result<IRMetalLibBinary>`. Same elsewhere.

Reply:
Postponing.